app/controllers/api/reservations_controller.rb

```diff
@@ -0,0 +1,70 @@
+class Api::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show update destroy]
+
+  # GET /api/reservations
+  def index
+    user_id = params[:user_id]
+    @reservations = Reservation.where(user_id:).includes(:motor)
```

Comment:
This is why your user not found spec test is not working. The problem is not the test, the problem is that it's getting motors even if user does not exist, hence returning an empty list with 200 response. Need error handling around the user first, like so:
```rb
@user = User.find_by(id: user_id)
if @user.nil?
  render json: { error: 'User not found' }, status: :not_found
  return
end
@reservations = @user.reservations.includes(:motor)
```


This is why error handling is necessary. Add error handling for all controllers where thiks issue may occur.